### ISSUE TITLE
Index can not be created when DS is too large.

### ISSUE THREAD

**Issue by marshyonline** [tech-greedy/singularity#165]
**Describe the bug**
An index can not be created when the Dataset is too large. 

```
# singularity index create COVID_mabs_2021
Response: 500
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>HTTPError: produced block is over 1MiB: big blocks can&#39;t be exchanged with other peers. consider using UnixFS for automatic chunking of bigger files, or pass --allow-big-block to override<br> &nbsp; &nbsp;at Object.errorHandler [as handleError] (/root/.nvm/versions/node/v16.16.0/lib/node_modules/@techgreedy/singularity/node_modules/ipfs-http-client/cjs/src/lib/core.js:84:15)<br> &nbsp; &nbsp;at runMicrotasks (&lt;anonymous&gt;)<br> &nbsp; &nbsp;at processTicksAndRejections (node:internal/process/task_queues:96:5)<br> &nbsp; &nbsp;at async Client.fetch (/root/.nvm/versions/node/v16.16.0/lib/node_modules/@techgreedy/singularity/node_modules/ipfs-utils/src/http.js:149:9)<br> &nbsp; &nbsp;at async Object.put (/root/.nvm/versions/node/v16.16.0/lib/node_modules/@techgreedy/singularity/node_modules/ipfs-http-client/cjs/src/dag/put.js:32:19)<br> &nbsp; &nbsp;at async IndexService.pinIndex (/root/.nvm/versions/node/v16.16.0/lib/node_modules/@techgreedy/singularity/dist/src/index/IndexService.js:64:39)<br> &nbsp; &nbsp;at async IndexService.pinIndex (/root/.nvm/versions/node/v16.16.0/lib/node_modules/@techgreedy/singularity/dist/src/index/IndexService.js:64:39)<br> &nbsp; &nbsp;at async IndexService.pinIndex (/root/.nvm/versions/node/v16.16.0/lib/node_modules/@techgreedy/singularity/dist/src/index/IndexService.js:64:39)<br> &nbsp; &nbsp;at async IndexService.pinIndex (/root/.nvm/versions/node/v16.16.0/lib/node_modules/@techgreedy/singularity/dist/src/index/IndexService.js:64:39)<br> &nbsp; &nbsp;at async IndexService.createIndexRequest (/root/.nvm/versions/node/v16.16.0/lib/node_modules/@techgreedy/singularity/dist/src/index/IndexService.js:143:25)</pre>
</body>
</html>
```

**Version**
`2.0.0`

**To Reproduce**
singularity index create *Large DS*

**Expected behavior**
Index is created.

This feature is super important to us to be able to provide our client a path for retrieval of their data while the Singularity WebUI is ready for consumption and retrievals

**Comment by marshyonline** [comment]
@xinaxu  - the tree for this DS i quite large, i have attached it. (Beware the tree is 17MiB)
[tree.txt](https://github.com/tech-greedy/singularity/files/10056053/tree.txt)

**Comment by xinaxu** [comment]
@marshyonline could you try this branch `xinaxu/index-service-fix` in #170 
Try reducing --max-links and --max-nodes if it does not work. - it may still be too large for a block if the filenames are too long.
```
Usage: singularity index create [options] <id_or_name>

Arguments:
  id_or_name                  The dataset id or name

Options:
  -l, --max-links <maxLinks>  Maximum number of links in each layer for a dynamic array or map (default: 1000)
  -n, --max-nodes <maxNodes>  The threshold above which should the structure be linked rather than embedded (default: 100)
  -m, --maxLink               The maximum number of links in each layer for a dynamic array or map
  -h, --help                  display help for command
```

**Comment by xinaxu** [comment]
Fixed in #170
